tests/testthat/test_array_manipulation.r

```diff
@@ -0,0 +1,327 @@
+  
+  # Reset limits
+  if (getOption('repr.matrix.max.rows') != orig_rows_limit) {
+    options('repr.matrix.max.rows' = orig_rows_limit)
+  }
+  if (getOption('repr.matrix.max.cols') != orig_cols_limit) {
+    options('repr.matrix.max.cols' = orig_cols_limit)
+  }
+})
+
+
+test_that("ellip.limit.arr limits arrays that are long and wide", {
+  ellip.h <- '\u22EF'
+  ellip.v <- '\u22EE'
+  ellip.d <- '\u22F1'
+  ellipses <- c(ellip.h, ellip.v, ellip.d)
+  # Make sure the limits are reasonable before we test.
+  orig_rows_limit <- getOption('repr.matrix.max.rows')
+  orig_cols_limit <- getOption('repr.matrix.max.cols')
+  
+  
+  # Make a 7x7 because I want to test with limits of 4, 5 and 6. I want to test 
+  # both the normal cases and the weird case where a dimension is one less than
+  # the limit (and therefore the 'smaller' output array is actually the same dim
+  # as the original)
+  test_mat <- matrix(1:49, ncol = 7)
+  test_df <- as.data.frame(test_mat)
+  test_dt <- as.data.table(test_mat)
+  
+  # We'll test with even and odd limits (but not all combinations of the two)
+  # Test with small numbers to keep things reasonable.
+  options('repr.matrix.max.rows' = 4L)
+  options('repr.matrix.max.cols' = 4L)
+  limited_mat <- ellip.limit.arr(test_mat)
+  limited_df <- ellip.limit.arr(test_df)
+  limited_dt <- ellip.limit.arr(test_dt)
+  expected_mat <- matrix(c('1', '2', ellip.v, '6', '7', '8', '9', ellip.v, '13',
+    '14', ellip.h, ellip.h, ellip.d, ellip.h, ellip.h, '36', '37',
+    ellip.v, '41', '42', '43', '44', ellip.v, '48', '49'), nrow = 5L)
+  expected_df <- as.data.frame(expected_mat)
+  expected_df[, 3] <- factor(expected_df[, 3], levels = ellipses)
+  names(expected_df) <- c('V1', 'V2', ellip.h, 'V6', 'V7')
+  rownames(expected_df) <- c('1', '2', ellip.v, '6', '7')
+  # The code, as a shortcut, just converts data.tables to data.frames.
+  #expected_dt <- as.data.table(expected_df)
+  expected_dt <- expected_df
+  expect_equal(limited_mat, expected_mat)
+  expect_equal(limited_df,  expected_df)
+  expect_equal(limited_dt,  expected_dt)
+
+  options('repr.matrix.max.rows' = 5L)
+  options('repr.matrix.max.cols' = 5L)
+  limited_mat <- ellip.limit.arr(test_mat)
+  limited_df <- ellip.limit.arr(test_df)
+  limited_dt <- ellip.limit.arr(test_dt)
+  expected_mat <- matrix(c('1', '2', '3', ellip.v, '6', '7', '8', '9', '10', 
+    ellip.v, '13', '14', '15', '16', '17', ellip.v, '20', '21', ellip.h, 
+    ellip.h, ellip.h, ellip.d, ellip.h, ellip.h, '36', '37', '38', ellip.v,'41',
+    '42', '43', '44', '45', ellip.v, '48', '49'), nrow = 6L)
+  expected_df <- as.data.frame(expected_mat)
+  expected_df[, 4] <- factor(expected_df[, 4], levels = ellipses)
+  names(expected_df) <- c('V1', 'V2', 'V3', ellip.h, 'V6', 'V7')
+  rownames(expected_df) <- c('1', '2', '3', ellip.v, '6', '7')
+  # The code, as a shortcut, just converts data.tables to data.frames.
+  #expected_dt <- as.data.table(expected_df)
```

Comment:
This could also get a test (that it is a dataframe), but I guess that `expect_equal(...)` already tests that both are a dataframe?